File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.c
#include <stdio.h>
#include <string.h>
int main()
{
    char num[5];

    while (1)
    {
        scanf("%s", num);

        if (strcmp(num, "0") == 0)
        {
            break;
        }

        int len = strlen(num);
        int palindrom = 1;

        for (int i = 0; i < len / 2; i++)
        {
            if (num[i] != num[len - 1 - i])
            {
                palindrom = 0;
                break;
            }
        }

        if (palindrom)
        {
            printf("yes\n");
        }
        else
        {
            printf("no\n");
        }
    }

    return 0;
}

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            String num = sc.next();

            if (num.equals("0")) {
                break;
            }

            boolean palindrom = true;
            int len = num.length();

            for (int i = 0; i < len / 2; i++) {
                if (num.charAt(i) != num.charAt(len - 1 - i)) {
                    palindrom = false;
                    break;
                }
            }

            if (palindrom) {
                System.out.println("yes");
            } else {
                System.out.println("no");
            }
        }

        sc.close();

    }
}
